test/test_utils.py

```diff
@@ -112,6 +114,96 @@ def __len__(self):
         return 10
 
 
+class TestCheckpoint(TestCase):
+
+    # Test whether checkpoint is being triggered or not. For this, we check
+    # the number of times forward pass happens
+    def test_checkpoint_trigger(self):
+
+        class Net(nn.Module):
+
+            def __init__(self):
+                super(Net, self).__init__()
+                self.count = 0
+
+            def forward(self, input_var):
+                self.count += 1
+                return input_var
+
+        # checkpointed
+        chunks = 2
+        module1, module2 = Net(), Net()
+        assert(module1.count == 0)
+        modules = [module1, module2]
+        input_var = torch.randn(3, 4, requires_grad=True)
+        out = checkpoint_sequential(modules, chunks, input_var)
+        out.sum().backward()
+        assert (module1.count == 2)
```

Comment:
You go through all this rigamarole to set up `module2`, and then you don't assert on it :) It'll be `count == 1`, right?